api/grpc_service.proto

```diff
@@ -78,4 +78,15 @@ message GrpcService {
   // A set of credentials that will be composed to form the `channel credentials
   // <https://grpc.io/docs/guides/auth.html#credential-types>`_.
   repeated Credentials credentials = 4;
+
+  // The human readable prefix to use when emitting statistics for the gRPC
+  // service.
+  //
+  // .. csv-table::
+  //    :header: Name, Type, Description
+  //    :widths: 1, 1, 2
+  //
+  //    streams_total, Counter, Total number of streams opened
+  //    streams_closed_<gRPC status code>, Counter, Total streams closed with <gRPC status code>
+  string stat_prefix = 5 [(validate.rules).string.min_bytes = 1];
```

Comment:
Since Envoy gRPC already has this, do we need this here vs. in Google gRPC?

Reply:
Two reasons to put it here:

1. Envoy gRPC doesn't have the `grpc-status` stats, only the HTTP status codes (which are 200 in most cases for gRPC, even on failure).
2. By having common stats in both clients, it simplifies integration tests (and by the same reasoning, operations).

Reply:
I would actually argue that we should add dynamic gRPC stats to the router stats that are output, which would be useful for all operations, even pass-through proxy. If we do end up adding that, this would become redundant. Thoughts?

Reply:
If we do that, then (1) is no longer relevant, but what about (2)?

I think longer term we'll want to better integrate with the gRPC clients own Census based stats (https://github.com/bazelbuild/bazel/blob/master/third_party/grpc/src/core/statistics/census_rpc_stats.h). If we do this, then we will definitely be duplicating stuff that appears in the upstream cluster or router stats. That would argue for us making this specialized to Google gRPC.

I'm leaning towards making this specific to Google gRPC as a result, LMK if we have design agreement and I'll make the change.

Reply:
+1